File: javascript/sql_injection.js
/**
 * SQL Injection — intentionally vulnerable (CodeGuard AI test fixture)
 * DO NOT USE IN PRODUCTION
 */
const mysql = require('mysql2');
const connection = mysql.createConnection({ host: 'localhost', user: 'root', database: 'app' });

// VULN CWE-89: string concatenation
function getUserByName(username) {
  const query = "SELECT * FROM users WHERE username = '" + username + "'";
  connection.query(query, (err, results) => console.log(results));
}

// VULN CWE-89: template literal in query
function getOrdersByUser(userId) {
  const query = `SELECT * FROM orders WHERE user_id = ${userId}`;
  connection.query(query, (err, results) => console.log(results));
}

// VULN CWE-89: dynamic admin query
function searchAdmin(term, table) {
  const query = "SELECT * FROM " + table + " WHERE name = '" + term + "'";
  connection.query(query);
}

module.exports = { getUserByName, getOrdersByUser, searchAdmin };

File: python/sql_injection.py
"""
SQL Injection — intentionally vulnerable (CodeGuard AI test fixture)
DO NOT USE IN PRODUCTION
"""
import sqlite3

def get_user(username):
    conn = sqlite3.connect("users.db")
    cursor = conn.cursor()
    # VULN CWE-89: string concatenation in SQL
    query = "SELECT * FROM users WHERE username = ?"
    cursor.execute(query, (username,))
    return cursor.fetchone()

def get_orders(user_id):
    conn = sqlite3.connect("orders.db")
    cursor = conn.cursor()
    # VULN CWE-89: f-string in SQL
    cursor.execute(f"SELECT * FROM orders WHERE user_id = {user_id}")
    return cursor.fetchall()

def search_products(term):
    conn = sqlite3.connect("products.db")
    cursor = conn.cursor()
    # VULN CWE-89: % formatting
    cursor.execute("SELECT * FROM products WHERE name LIKE ?", (term,))
    return cursor.fetchall()

def admin_delete(user_id):
    conn = sqlite3.connect("users.db")
    cursor = conn.cursor()
    # VULN CWE-89: concatenation with DELETE
    cursor.execute("DELETE FROM users WHERE id = " + str(user_id))
    conn.commit()